trieste/models/optimizer.py

```diff
@@ -151,6 +151,27 @@ def train_fn() -> None:
             train_fn()
 
 
+@dataclass
+class KerasOptimizer:
+    """Optimizer wrapper for training neural network models implemented with Keras."""
+
```

Comment:
Are these different to the tf optimizers? Sorry for my ignorance

Reply:
eh, this is a bit messy - underlying optimizers are the same, I could work with BatchOptimizer but its a bit silly to use it as we use Keras' fit method rather then the optimizer directly where we construct our own batches etc, I thought its better to avoid a bit of confusion and create a wrapper for Keras with some additional arguments (loss and metrics) - I do run into trouble then with configs bcs you cannot map the same optimizers to multiple wrappers :|